Core/include/Acts/Vertexing/AdaptiveGridDensityVertexFinder.ipp

```diff
@@ -44,6 +44,13 @@ auto Acts::AdaptiveGridDensityVertexFinder<trkGridSize, vfitter_t>::find(
         }
         continue;
       }
+      if (trkParams.referenceSurface()
```

Comment:
You're not doing a conversion here so this is probably not very expensive. I think you could directly call `.center(gctx)` instead of manually getting the translation from the transform matrix.